addons/account/models/account_journal_dashboard.py

```diff
@@ -524,44 +524,36 @@ def _fill_sale_purchase_dashboard_data(self, dashboard_data):
             "account_move.company_id",
         ]
         payment_field_list = [
-            "account_move_line.journal_id",
-            "account_move_line.move_id",
-            "ABS(account_move_line.amount_residual) AS amount_total_company",
+            SQL("account_move_line.journal_id"),
+            SQL("account_move_line.company_id"),
+            SQL("account_move_line.date_maturity < %s AS late", fields.Date.context_today(self)),
+            SQL("SUM(account_move_line.amount_residual) AS amount_total_company"),
+            SQL("COUNT(*)"),
         ]
         # DRAFTS
         query, params = sale_purchase_journals._get_draft_sales_purchases_query().select(*bills_field_list)
         self.env.cr.execute(query, params)
         query_results_drafts = group_by_journal(self.env.cr.dictfetchall())
 
-        # WAITING BILLS AND PAYMENTS
+        # WAITING AND LATE BILLS AND PAYMENTS
         query_results_to_pay = {}
+        late_query_results = {}
         if purchase_journals:
```

Comment:
Could we not factorize the two if also since there is a lot of duplicated codes ?